### DISCUSSION TITLE
Không activate được conda env trong VsCode

### DISCUSSION THREAD

**Discussion by tungdop2** [Foundation-Learning-UETAILab/Week-0-Introduction#14] [Q&A] [answered]
Em có thể activate conda env ở terminal.
![image](https://user-images.githubusercontent.com/69032433/142186179-3283ca2c-561a-4324-b0d8-620fe04cd6e8.png)
Nhưng trong terminal của Vscode lại không được, tại sao vậy ạ?
![image](https://user-images.githubusercontent.com/69032433/142186319-51ce7e28-4802-48d2-a93d-00c3befea99d.png)

**Comment by HaiNguyen2903** [comment]
Anh nghĩ khả năng cao là em chưa setup PYTHONPATH trong vscode của mình.
Suggest em thử Ctrl+Shift+P chọn `Select Python Interpreter` là env của em thử xem có work không?

**Reply by nmd2k** [reply]
Này đã solve được vấn đề của @tungdop2 chưa 🚀

**Reply by tungdop2** [reply]
em có Select Python Interpreter nhưng trong vẫn ko đc ạ

**Comment by nmd2k** [comment, marked as answer]
Có lẽ cái lỗi này không nằm ở Vscode, bên conda nó support trên 1 vài cái shell và trong Windows thì có tới tận 2 loại shell là `cmd.exe` và `power shell`. Khả năng cao là do bạn (hoặc nó auto) `init` với 1 shell và shell kia chưa được init nên `conda activate <env_name>` không hoạt động ở đây.

Bạn thử mặc định cái terminal của bạn dùng `cmd.exe` hay `powershell` rồi activate nó xem.

###  Conda command to init shell
```bash
$ conda init <SHELL_NAME>
```
Currently supported shells are:
  - bash
  - cmd.exe
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell
 
###  Cách thay đổi terminal mặc định trong vscode:
- Press **Ctrl + Shift + P** to open the command palette.
- Type **profile** in the searcher.
- Select **Terminal: Select Default Profile**.
- Select 'Command Prompt' or 'PowerShell'
- The next time you try to open the terminal you should see 'CMD' instead of 'PowerShell'.
